utils/hashing-proc-macro/Cargo.toml

```diff
@@ -0,0 +1,21 @@
+[package]
+name = "hashing-proc-macro"
+version = "4.0.0-dev"
+authors = ["Parity Technologies <admin@parity.io>"]
+edition = "2018"
+license = "Apache-2.0"
+homepage = "https://substrate.dev"
+repository = "https://github.com/paritytech/substrate/"
+description = "This crate provides procedural macros for calculating static hash."
+documentation = "https://docs.rs/hashing-proc-macro"
+
+[package.metadata.docs.rs]
+targets = ["x86_64-unknown-linux-gnu"]
+
+[lib]
+proc-macro = true
+
+[dependencies]
+syn = { version = "1.0.77", features = ["full", "parsing"] }
+blake2-rfc = { version = "0.2.18", default-features = false }
+twox-hash = { version = "1.5.0", default-features = false }
```

Comment:
we can't rely on sp-core directly here ? or is it a bad idea ?

Reply:
Sounds good to avoid redundant code, on the other hand, core does have a few/many deps, I don't think there is a cyclic dep with current changes, but can be a bit of a footgun, not sure.
Maybe pulling the hashing code in another utils/hashing crate that this proc macro and sp_core can depend on may be better?

Reply:
yes having sp-core-hashing seems the best to me. But at the same time it is a duplication of only ~100 lines of code. But maybe those lines will incease in the future.